cpp/src/Slice/Grammar.y

```diff
@@ -385,6 +385,75 @@ module_def
         $$ = nullptr;
     }
 }
+| ICE_MODULE ICE_SCOPED_IDENTIFIER
+{
+    auto ident = dynamic_pointer_cast<StringTok>($2);
```

Comment:
Can ICE_SCOPED_IDENTIFIER start with `::`?

Reply:
Yes it can, but if you type: `::Hello::There`, we'll pick up 3 modules: `["", "Hello", "There"]`.
And `""` is not a valid module identifier, so it fails.

Reply:
But I think you're right, I'll add an extra so we can emit a better message than `illegal identifier: `'`